library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -423,11 +466,15 @@ public void emitReplayedTuples(PreparedStatement ps)
     List<Partition<AbstractJdbcPollInputOperator<T>>> newPartitions = new ArrayList<>(
         getPartitionCount());
 
-    HashMap<Integer, KeyValPair<Integer, Integer>> partitionToRangeMap = null;
+    final List<KeyValPair<Integer, Integer>> partitionRanges;
     try {
       store.connect();
       dslContext = createDSLContext();
-      partitionToRangeMap = getPartitionedQueryRangeMap(getPartitionCount());
+      if (getPartitionCount() > 0) {
+        partitionRanges = getPartitionedQueryRanges(getPartitionCount());
+      } else {
+        partitionRanges = new ArrayList<>(0);
+      }
```

Comment:
Looks better if this check is moved into getPartitionedQueryRanges; also makes that function more robust by avoiding divide-by-zero errors in future maintenance changes.